File: youlexuan-manager-web/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.List;
import java.util.Map;

@Controller
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    @ResponseBody
    public List<TbBrand> findAll(){
        return brandService.getAll();
    }


    @RequestMapping("/search")
    @ResponseBody
    private PageResult fandPage(@RequestBody TbBrand tbBrand, int page, int rows) {
        PageResult pageResult = brandService.findpage(tbBrand, page, rows);
        return pageResult;
    }

    @RequestMapping("/add")
    @ResponseBody
    public Result add(@RequestBody TbBrand tbBrand){
        try {
            brandService.add(tbBrand);
            return new Result(true,"添加成功");
        }catch (Exception e){
            return new Result(true,"添加失败");
        }
    }

    @RequestMapping("/findOne")
    @ResponseBody
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/update")
    @ResponseBody
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            return new Result(true,"修改失败");
        }
    }

    @RequestMapping("/deleteId")
    @ResponseBody
    public Result deleteId(Long[] ids){
        try {
            brandService.deleteId(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            return new Result(true,"删除失败");
        }
    }

    @RequestMapping("/selectOptionList")
    @ResponseBody
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}

File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("branController",function ($scope,$controller,brandService) {

    //继承baseController
    $controller('baseController',{$scope:$scope});

    //查询所有类
    $scope.findAll=function () {
        brandService.findAll().success(function (response) {
            $scope.list=response;
        })
    };

    //分页模糊查询
    $scope.searchEntity={};
    $scope.search=function(page,rows){
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;
            })
    };


    //根据id查找类
    $scope.findone=function(id){
        brandService.findone(id).success(
            function (response) {
                $scope.entity=response;
            }
        )
    };

    //添加与修改
    $scope.save=function () {
        if($scope.entity.id!=null){
            brandService.update($scope.entity).success(
                function (response) {
                    if (response.success){
                        $scope.reloadList();
                    }else {
                        alert(response.message)
                    }

                }
            )
        }else{
            brandService.add($scope.entity).success(
                function (response) {
                    if (response.success){
                        $scope.reloadList();
                    }else {
                        alert(response.message)
                    }

                }
            )
        }

    };

    //根据id删除类
    $scope.deleteId=function () {
        brandService.deleteId($scope.selectIds).success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }

            }
        )

    };
})
